android/modules/ui/src/java/ti/modules/titanium/ui/widget/picker/TiUITimePicker.java

```diff
@@ -160,15 +174,36 @@ public void setValue(long value, boolean suppressEvent)
 	@Override
 	public void onTimeChanged(TimePicker view, int hourOfDay, int minute)
 	{
+		detectTimeChange();
+	}
+	
+	private void detectTimeChange() {
 		Calendar calendar = Calendar.getInstance();
-		calendar.set(Calendar.HOUR_OF_DAY, hourOfDay);
-		calendar.set(Calendar.MINUTE, minute);
-		if (!suppressChangeEvent) {
+		TimePicker picker = (TimePicker) getNativeView();
+		boolean hasChanged;
+		if (Build.VERSION.SDK_INT >= 23) {
+			hasChanged = lastSelectedHour != picker.getHour() || lastSelectMin != picker.getMinute();
```

Comment:
These warnings are only thrown from Dalvik which is why I didn't see them (and won't be seen on Android 5.0+ due to moving to ART). These warnings are expected behaviour, but ugly. Since the references are properly guarded no issue will actually come from this.

If we do want to address this we should do it with a class in the `appcompat` module instead of a nested class. Also note that it doesn't provide any additional safety as it still relies on the
```JS
if (Build.VERSION.SDK_INT ...
```
statement. It's just preventing Dalvik from seeing the broken reference.